src/app/core/store/reducers/defendants.reducer.ts

```diff
@@ -62,6 +62,60 @@ export function reducer(
       } as DefendantsState;
     }
 
+    case fromDefendants.DefendantsActionTypes.UpdateDefendant: {
+
+      const defendant = action.payload;
+
+      return {
+        ...state,
+        loading: {
+          ...state.loading,
+          [defendant.id]: true,
+        },
+        loaded: {
+          ...state.loaded,
+          [defendant.id]: false,
+        },
+      }
+    }
+
+    case fromDefendants.DefendantsActionTypes.UpdateDefendantSuccess: {
+
+      const defendant = action.payload;
+
+      return {
+        ...state,
+        entities: {
+          ...state.entities,
+          [defendant.id]: defendant,
```

Comment:
You must call `.toJson()` on the defendant. Storing instances of objects inside the store is a really bad practice that usually ends up with weird bugs due to the state being modified outside of the reducer.